src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-form/configuration-form.component.ts

```diff
@@ -0,0 +1,191 @@
+        '{3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:' +
+        '((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:' +
+        '[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+        '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*';
+      validators.push(Validators.pattern(ipv4Ipv6Rgx));
+    }
+
+    if (configOption.type === 'uuid_d') {
+      validators.push(
+        Validators.pattern('[a-z0-9]{8}\\-[a-z0-9]{4}\\-[a-z0-9]{4}\\-[a-z0-9]{4}\\-[a-z0-9]{12}')
+      );
+    }
+
+    return validators;
+  }
+
+  setResponse(response: ConfigFormModel) {
+    this.response = response;
+    const validators = this.getValidators(response);
+
+    this.configForm.get('name').setValue(response.name);
+    this.configForm.get('desc').setValue(response.desc);
+    this.configForm.get('long_desc').setValue(response.long_desc);
+    this.configForm.get('default').setValue(response.default);
+    this.configForm.get('daemon_default').setValue(response.daemon_default);
+    this.configForm.get('services').setValue(response.services);
+
+    if (this.response.value) {
+      this.response.value.forEach((value) => {
+        // Check value type. If it's a boolean value we need to convert it because otherwise we
+        // would use the string representation. That would cause issues for e.g. checkboxes.
+        let sectionValue = null;
+        if (value.value === 'true') {
+          sectionValue = true;
+        } else if (value.value === 'false') {
+          sectionValue = false;
+        } else {
+          sectionValue = value.value;
+        }
+        this.configForm
+          .get('values')
+          .get(value.section)
+          .setValue(sectionValue);
+      });
+    }
+
+    this.availSections.forEach((section) => {
+      this.configForm
+        .get('values')
+        .get(section)
+        .setValidators(validators);
+    });
+
+    this.type = response.type;
+  }
+
+  createRequest(): ConfigFormCreateRequestModel {
+    const value = [];
+
+    this.availSections.forEach((section) => {
+      const sectionValue = this.configForm.getValue(section);
+      value.push({ section: section, value: sectionValue });
+    });
+
+    const request = new ConfigFormCreateRequestModel();
+    request.name = this.configForm.getValue('name');
+    request.value = value;
+    return request;
+  }
+
+  createAction() {
+    const request = this.createRequest();
+    this.configService.create(request).subscribe(
+      () => {
+        this.notificationService.show(
+          NotificationType.success,
+          'Config option ' + request.name + ' has been updated.',
+          'Update config option'
+        );
+        this.router.navigate(['/configuration']);
+      },
+      () => {
+        this.configForm.setErrors({ cdSubmitButton: true });
+      }
+    );
+  }
+
+  submit() {
+    if (this.configForm.pristine) {
```

Comment:
This is a good ideia, but should be implemented on the SubmitButton component.
I have created an issue for it: : https://tracker.ceph.com/issues/34530
So you can removed it from here :)

Reply:
Okay, will do. Thanks.

Reply:
Done!